explorer/models.py

```diff
@@ -147,6 +149,78 @@ class Meta:
 
 
 class QueryResult(object):
+    def get_type_code_and_column_indices_to_be_masked_dict(self):
+        """
+        Returns a dictionary of type code and column indices to be masked
+        Return type:
+        {
+            type_code: [column indices that match the type code]
+        }
+        Eg.
+        Say a table has three fields id, data, random_text. id is of type INT, data is of type JSON, and random_text is of type TEXT.
+        Then the return value will be:
+        {
+            TYPE_CODE_FOR_JSON: [1],
+            TYPE_CODE_FOR_TEXT: [2]
+        }
+        as 1 is the column index for JSON and 2 is the column index for TEXT
+        """
+        type_code_and_column_indices_to_be_masked_dict = {
+            TYPE_CODE_FOR_JSON: [],
+            TYPE_CODE_FOR_TEXT: []
+        }
+        # Collect the indices for JSON and text columns
+        for index, column in enumerate(self._description):
+            if hasattr(column, "type_code") and column.type_code in type_code_and_column_indices_to_be_masked_dict:
+                type_code_and_column_indices_to_be_masked_dict[column.type_code].append(index)
+
+        return type_code_and_column_indices_to_be_masked_dict
+
+    def mask_json_data(self, data):
+        return json.dumps(replace_regex(data, PII_MASKING_PATTERN_REPLACEMENT_DICT))
+
+    def mask_text_data(self, data):
+        return replace_regex(data, PII_MASKING_PATTERN_REPLACEMENT_DICT)
```

Comment:
why do we have to pass it here and not use directly in utils methods? 
`PII_MASKING_PATTERN_REPLACEMENT_DICT`